src/pybind/mgr/volumes/fs/operations/group.py

```diff
@@ -81,7 +81,7 @@ def list_snapshots(self):
         try:
             dirpath = os.path.join(self.path,
                                    self.vol_spec.snapshot_dir_prefix.encode('utf-8'))
-            return listdir(self.fs, dirpath)
+            return listsnaps(self.fs, self.vol_spec, dirpath, filter_inherited_snaps=True)
```

Comment:
maybe sufficient to pass snap_dir_prefix?
where snap_dir_prefix = self.vol_spec.snapshot_dir_prefix.encode('utf-8')

Reply:
I thought so, but since this is an API, giving the control to user to pass both validation strings didn't sound correct to me.